MySQL 8.0.1 removes support for the `my_bool` datatype

This issue only affects MySQL; MariaDB will continue to support `my_bool`.

The following build error occurs when attempting to compile with MySQL 8.0.1:
```
motion.c: In function ‘motion_init’:
motion.c:1146:13: error: unknown type name ‘my_bool’; did you mean ‘bool’?
 1146 |             my_bool my_true = TRUE;
      |             ^~~~~~~
      |             bool
make: *** [Makefile:105: motion.o] Error 1
make: *** Waiting for unfinished jobs....
```

Noted in 8.0.1 changelog (https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-1.html#mysqld-8-0-1-compiling):

The **mysql.h** header file now requires a C++ or C99 compiler to
compile.

The `my_bool` type is no longer used in MySQL source code. Any third-party code that used this type to represent C boolean variables should use the bool or int C type instead.

--- a/src/motion.c
+++ b/src/motion.c
@@ -1047,7 +1047,7 @@ static int dbse_init_mysql(struct context *cnt){
                 return -2;
             }
             #if (defined(MYSQL_VERSION_ID)) && (MYSQL_VERSION_ID > 50012)
-                my_bool my_true = TRUE;
+                int my_true = TRUE;
                 mysql_options(cnt->database, MYSQL_OPT_RECONNECT, &my_true);
             #endif
         }

